core-bundle/src/Resources/contao/models/ContentModel.php

```diff
@@ -432,6 +433,42 @@ public static function countPublishedByPidAndTable($intPid, $strParentTable, arr
 
 		return static::countBy($arrColumns, array($intPid, $strParentTable), $arrOptions);
 	}
+
+	/**
+	 * Get the default meta data or null if not applicable.
+	 */
+	public function getMetaData(): ?MetaData
+	{
+		// Ignore if `overwriteMeta` isn't set
+		if ('' === $this->overwriteMeta)
```

Comment:
I think so. The meta data from `ContentModel->getMetaData()` is the one for the core use cases (image/gallery/news content elements)  where `overwriteMeta` is used. Otherwise you'd either have to write one of this all the time:
* ```php
  if('1' === $model->overwriteMeta) { // breaks fluent setting
    $builder->overwriteMeta($model->getMetaData())
  }
  ```
* ```php
  $builder
    ->...
    ->setMetaData('1' === $model->overwriteMeta ? $model->getMetaData() : null)
  ```

Reply:
Also: if the checkbox isn't set - the data of the widgets should ideally not be in the model anyways (but it is due to the limits of the current DCA/shared model/data handling concept).

Reply:
> is the one for the core use cases (image/gallery/news content elements)

Thats the problem I always have with it. There are different (core) use cases (e.g. `overwriteLink` and its related fields), and the generic `ContentModel` doesn't know about them. Unfortunately, I can't think of a simple solution for this.

I don't see a problem with breaking the fluent setter for explicit programming. But it's kinda true that if `overwriteMeta` is not set, then there actually shouldn't be any metadata for the current element 🤔 

=> How about we rename this to `getOverwriteMetaData` to make it more obvious?

Reply:
> There are different (core) use cases (e.g. overwriteLink and its related fields)

Imo `overwriteLink` should be handled by the download content element (it's the only one using it). `overwriteMeta` is used by image, text, accordionSingle.. So I'm not sure where this should go. Repeating the code over and over again should not be the goal. Maybe we could have a trait for the content elements, though? :thinking: 

> => How about we rename this to getOverwriteMetaData to make it more obvious?

Good point. Changed in fd4a167686a765b11b03f197cd46a626bb2b21f9 as long as we do not have a better solution.